remote-config/src/test/kotlin/com/rakuten/tech/mobile/remoteconfig/ConfigCacheSpec.kt

```diff
@@ -125,11 +126,45 @@ class ConfigCacheSpec : RobolectricBaseSpec() {
 
         When calling stubVerifier.verify(any()) itReturns false
 
-        val cache = ConfigCache(stubFetcher, createFile(filename), stubPoller, stubVerifier)
+        val cache = ConfigCache(stubFetcher, createFile(filename), stubPoller, stubVerifier, false)
 
         cache["foo"] shouldEqual null
     }
 
+    @Test
+    fun `should not apply after fetching`() {
+        val filename = "cache.json"
+
+        When calling stubFetcher.fetch() itReturns createConfig(hashMapOf("foo" to "bar"))
+        When calling stubVerifier.verify(any()) itReturns true
+        val cache = ConfigCache(stubFetcher, createFile(filename), AsyncPoller(1), stubVerifier, false)
+
+        CoroutineScope(Dispatchers.Main).launch {
+            withContext(Dispatchers.Default) {
+                val cache = ConfigCache(stubFetcher, createFile(filename), AsyncPoller(1),
+                        stubVerifier, true)
+                delay(1000)
```

Comment:
When using `runBlockingTest`, you can use `advanceTimeBy` so that async tests don't make the execution time longer: https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-test/

Reply:
used `advanceTimeBy`